ipaserver/rpcserver.py

```diff
@@ -493,7 +493,7 @@ def marshal(self, result, error, _id=None,
 
     def unmarshal(self, data):
         try:
-            d = json.loads(data)
+            d = json.loads(ipautil.decode_json(data))
```

Comment:
You are assuming that bytes always means UTF-8 encoded text. Although it's the common case for JSON, it is not necessarily correct. JSON can also be encoded as UTF-16 LE/BE or UTF-32 LE/BE.